models/vogels_sprekeler_connection.h

```diff
@@ -298,8 +298,7 @@ VogelsSprekelerConnection< targetidentifierT >::set_status(
   updateValue< double >( d, names::Kplus, Kplus_ );
 
   // check if weight_ and Wmax_ has the same sign
-  if ( not( ( ( weight_ >= 0 ) - ( weight_ < 0 ) )
-         == ( ( Wmax_ >= 0 ) - ( Wmax_ < 0 ) ) ) )
+  if ( weight_ != 0 && not( ( weight_ > 0 ) == ( Wmax_ > 0 ) ) )
```

Comment:
@sanjayankur31 The comment above the check is on entirely in sync with the code. Also, the code needs quite a bit of thought to understand it. Since we now allow C++11 constructs in NEST, I would suggest
```c++
weight != 0 and ( std::signbit( weight_ ) != std::signbit( Wmax_ ) )
```